server/src/main/java/org/elasticsearch/action/search/ExpandSearchPhase.java

```diff
@@ -100,6 +100,10 @@ private void doRun() {
                     if (hit.getInnerHits() == null) {
                         hit.setInnerHits(Maps.newMapWithExpectedSize(innerHitBuilders.size()));
                     }
+                    if (hit.isPooled() == false) {
+                        // TODO: make this work pooled by forcing the hit itself to become pooled as needed here
+                        innerHits = innerHits.asUnpooled();
+                    }
```

Comment:
ARG!!!! This indeed is frustrating. But, I agree, we should do the dumb thing first to fix 8.15.x and do something nicer & more performant later.